rudder-rest/src/main/scala/com/normation/rudder/rest/RoleApiMapping.scala

```diff
@@ -44,14 +44,138 @@ import com.normation.rudder.api.ApiAclElement
 import com.normation.rudder.Role
 import com.normation.rudder.api.AclPathSegment
 
+/*
+ * The goal of that class is to map Authorization to what API
+ * they give access.
+ * At some point, we will need AuthorizationType to be extensible, so
+ * that a new module can provide new authorization. For now, a plugin
+ * need to map its authorizations on the existing ones.
+ */
+trait AuthorizationApiMapping {
+  def mapAuthorization(authz: AuthorizationType): List[ApiAclElement]
+}
+
+/*
+ * An extensible mapper that allows for plugins to contribute to
+ * its mapper
+ */
+class ExtensibleAuthorizationApiMapping(base: List[AuthorizationApiMapping]) extends AuthorizationApiMapping {
+
+  private[this] var mappers: List[AuthorizationApiMapping] = base
+
+  def addMapper(mapper: AuthorizationApiMapping): Unit = {
+    // no need to add again and again the default mapper - it's ok, we have it.
+    if(mapper != AuthorizationApiMapping.OnlyAdmin) {
+      mappers = mappers :+ mapper
+    }
+  }
+
+  override def mapAuthorization(authz: AuthorizationType): List[ApiAclElement] = {
+    mappers.flatMap( _.mapAuthorization(authz))
+  }
+}
+
+
+object AuthorizationApiMapping {
+  implicit class ToAuthz(api: EndpointSchema) {
+    def x: ApiAclElement = AuthzForApi(api)
+  }
+
+  /*
+   * A default mapping for "only 'all rights' (ie admin) can access it
+   */
+  case object OnlyAdmin extends AuthorizationApiMapping {
+    override def mapAuthorization(authz: AuthorizationType): List[ApiAclElement] = Nil
+  }
+
+  /*
+   * The core authorization/api mapping, ie the authorization for Rudder
+   * default API.
+   */
+  object Core extends AuthorizationApiMapping {
+
+    override def mapAuthorization(authz: AuthorizationType): List[ApiAclElement] = {
+      import AuthorizationType._
+      // shorthand to get authz for a given api
+
+      authz match {
+        case NoRights             => Nil
+        case AnyRights            => ApiAuthz.allAuthz.acl
+        // Administration is Rudder setting
+
+        case Administration.Read  => SettingsApi.GetAllSettings.x :: SettingsApi.GetSetting.x :: Nil
+        case Administration.Write => SettingsApi.ModifySettings.x :: SettingsApi.ModifySetting.x :: Nil
+        case Administration.Edit  => SettingsApi.ModifySettings.x :: SettingsApi.ModifySetting.x :: Nil
+
+        case Compliance.Read      => ComplianceApi.GetGlobalCompliance.x :: ComplianceApi.GetRulesCompliance.x ::
+                                     ComplianceApi.GetRulesComplianceId.x :: ComplianceApi.GetNodesCompliance.x ::
+                                     ComplianceApi.GetNodeComplianceId.x :: Nil
+        case Compliance.Write     => Nil
+        case Compliance.Edit      => Nil
+
+        // Configuration doesn't give API rights.
+        // But as nothing manage parameter API, I think it's the correct place
+        // to add it.
+        case Configuration.Read   => ParameterApi.ListParameters.x :: ParameterApi.ParameterDetails.x :: Nil
```

Comment:
I though Configuration was giving access to all configs (Rules/Directives/Parameters/Techniques), am i wrong ?
So it should provide access to all other element not only parameter

Reply:
In fact it's already like that in 4.3